### PR TITLE
Add confirmation dialog for load sample products

### LINKED ISSUE #33167: Add confirmation for load sample products
## User Story

As a new merchant, I want to be warned about sample products loading images from woocommerce.com, so that I can make an informed decision.

## Acceptance Criteria

* Add a confirmation dialog before loading the sample products stating "We'll import images from woocommerce.com to set up your sample products."
* Buttons should be "Import sample products" (primary) and "Cancel" (secondary).

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #33167.

This PR adds a confirmation dialog before loading sample products

**Acceptance Criteria**
- Add a confirmation dialog before loading the sample products stating `"We'll import images from woocommerce.com to set up your sample products."`
- Buttons should be `"Import sample products"` (primary) and `"Cancel"` (secondary).

Screen Shot 2022-05-24 at 15 47 40

### How to test the changes in this Pull Request:

Make sure your site has the `woocommerce_allow_tracking` option set to yes

**Test load sample product - stack layout**

1. Go to Tools > WCA Test Helper > Experiments
2. Enable treatment for `woocommerce_products_task_layout_stacked`
3. Make sure there are no products in store (if there are in trash, empty them)
4. Navigate to `WooCommerce > Home`
5. Go to "Add products" task
6. Click `View more product types`
7. Click `Load sample products`
8. Observe that a confirmation dialog  is displayed
9. Click `Import sample products` button
10. Confirm that it loads the sample products
11. Go back to "Add products" task
12. Repeat 6~8 but click the `Cancel` button instead.
13. Confirm that the confirmation dialog is hidden

**Test load sample product - card layout**

1. Go to Tools > WCA Test Helper > Experiments
2. Enable treatment for `woocommerce_products_task_layout_card`
3. Make sure there are no products in store (if there are in trash, empty them)
4. Navigate to `WooCommerce > Home`
5. Go to "Add products" task
6. Click `View more product types`
7. Click the `CAN’T DECIDE?` card
8. Observe that a confirmation dialog  is displayed
9. Click `Import sample products` button
10. Confirm that it loads the sample products
11. Go back to "Add products" task
12. Repeat 6~8 but click the `Cancel` button instead.
13. Confirm that the confirmation dialog is hidden

**Test load sample product - import product view**

1. Go to Tools > WCA Test Helper > Experiments
2. Enable treatment for either `woocommerce_products_task_layout_stacked` or `woocommerce_products_task_layout_card`
3. Make sure there are no products in store (if there are in trash, empty them)
4. Go to OBW
5. Choose `Yes, on another platform` on the Business Details step
6. Navigate to `WooCommerce > Home`
7. Go to "Add products" task
8. Click `Or add your products from scratch`
9. Click the `Import sample products` button
10. Observe that a confirmation dialog  is displayed
11. Click `Import sample products` button
12. Confirm that it loads the sample products
13. Go back to "Add products" task
14. Repeat 6~8 but click the `Cancel` button instead.
15. Confirm that the confirmation dialog is hidden

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm nx changelog <project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33171/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Add translator domain to texts` 773ab10c61122e6a8207a3ba2e1ccb1bf6c830ee
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+export const LoadSampleProductConfirmModal: React.VFC< Props > = ( {
+	onCancel,
+	onImport,
+} ) => {
+	return (
+		<Modal
+			className="woocommerce-products-load-sample-product-confirm-modal"
+			title="Load sample products"
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx]: Missed translation here

File: plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			onRequestClose={ onCancel }
+		>
+			<Text className="woocommerce-confirmation-modal__message">
+				{ __(
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx:29]: The `__` calls are missing `woocommerce` 2nd parameter

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Thanks for such a quick turnaround on this one, @chihsuan! I have a few feedback:

For **Import screen** and **Stacked layout screen**, it seems `wcadmin_task_completion_time` track is triggered when clicking "Load sample products", making it inaccurate. We should change the track to trigger after user confirm to load samples.

**Comment by chihsuan** [comment]
> For Import screen and Stacked layout screen, it seems wcadmin_task_completion_time track is triggered when clicking "Load sample products", making it inaccurate. We should change the track to trigger after user confirm to load samples.

~Nice catch! Changed in 31af5fe5cc0a09855c9e036b3ca48a383b2af4fd~

Wait, looks like there is another track. 🤔

File: plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx]: Oops. Added in 8ff2c334d7144b390811a7c15b5ab80e8f54d707.

File: plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/client/tasks/fills/components/load-sample-product-confirm-modal.tsx:29]: Added in https://github.com/woocommerce/woocommerce/commit/8ff2c334d7144b390811a7c15b5ab80e8f54d707.

**Comment by chihsuan** [comment]
> For Import screen and Stacked layout screen, it seems wcadmin_task_completion_time track is triggered when clicking "Load sample products", making it inaccurate. We should change the track to trigger after user confirm to load samples.

@ilyasfoo Thanks for catching this. Looks like we have duplicated the `wcadmin_task_completion_time` track for import sample products. I just removed an unnecessary one.

https://github.com/woocommerce/woocommerce/pull/33171/commits/1b8e90b52975889934ac793537ec9b4349114eae

**Review by ilyasfoo — APPROVED** [review]
>Thanks for catching this. Looks like we have duplicated the wcadmin_task_completion_time track for import sample products. I just removed an unnecessary one.

@chihsuan Nice! That would've been a sneaky bug otherwise, thanks for fixing it!

LGTM! 🚢 🚢

**Comment by jonathansadowski** [comment]
This has been cherry-picked into the 6.6 release branch.

**Pull request merged**